### ISSUE TITLE
Provide pre-built binaries to run without Python

### ISSUE THREAD

**Issue by filips123** [xonsh/xonsh#3387]
Please provide pre-built binaries to run Xonsh without Python. Ideally, they should just be single file, from which you can directly access and use Xonsh. You can use PyInstaller for this and make build that files automatically on CI.

**Comment by BYK** [comment]
I tried using PyOxidizer for this: https://pyoxidizer.readthedocs.io/

Our liberal use of `__file__` is a deal-breaker for that though. I'll see if I can get a working version/

**Comment by scopatz** [comment]
Thanks @BYK! Again, any changes to support this kind of thing would be amazing

**Comment by astronouth7303** [comment]
I would say the tooling to do this is outside of the scope of xonsh. If there's good tools to do it, though, we should.

@ironfroggy's https://github.com/ironfroggy/feet is also a thing that exists to do this.

There should be a minimal use of `__file__` in the codebase? I've been using [xonsh as a zipapp](https://gitlab.com/doyouevenclick/xonsh) for a while now, and the only adaptions are around `xpip` and user-installed packages.

Because of my work with [PursuedPyBear](https://ppb.dev/), this is a problem space I've been mulling over. (https://github.com/pyz-dispenser/cpython-static/ is part of that.) One of the big problems is Mac: They don't really do single-file executables the same way that Windows and Linux do, and they're much bigger about code signing. Other problems come from system dependencies: It's easy enough to link libz and readline, but openssl is 1. A bear, 2. needs extra datafiles. On top of OpenSSL's problems with Windows and Mac. (See also [PEP 543](https://www.python.org/dev/peps/pep-0543/)). And excluding OpenSSL isn't really feasible because that means no HTTPS or other encrypted communications. Oh, and the MSVC Redistributable.

Basically, this is a messy problem.

**Comment by anki-code** [comment]
Hi @filips123!
Here is my solutions:
1. [Xonsh AppImage as one executable file](https://xon.sh/appimage.html) you can download from [xonsh release`s assets](https://github.com/xonsh/xonsh/releases). It's not really pure binary it's a package with python inside (read more about AppImage technology) but it works stable and predictable.
2. [Xonsh as the one linux-portable binary file](https://github.com/xonsh/xonsh/issues/4016) that I built using Nuitka. You can try and continue this work.

**Comment by anki-code** [comment]
JFYI The simpler recipe to build xonsh binary with Nuitka is here - https://github.com/Nuitka/Nuitka/issues/1401#issue-1118685121
